mitodl/open-edx-plugins
# feat: added github workflow to run tests for all open-edx plugins on different edx distributions

### What are the relevant tickets?
Closes https://github.com/mitodl/open-edx-plugins/issues/212

### Description (What does it do?)
This PR adds GitHub Actions to run tests for all OpenEdx plugins on different edX distributions. The edX distributions on which tests run are quince, palm, olive, and master.

**Note:**
- The tests run with pytest directly. Although I attempted to run them using Pants, it failed due to their dependency on edx-platform.
- Only the edx-sysadmin plugin has tests. I searched for tests for other plugins but didn't find any.

### How can this be tested?

- Review the workflow logs to observe tests running.
- For local testing:
   - Clone this [branch](https://github.com/Anas12091101/open-edx-plugins/tree/github-actions) into the `src` folder in the edx workspace.
   - Modify paths from `/open-edx-plugins` to `/edx/src/open-edx-plugins` in the [run_devstack_integration_tests.sh](https://github.com/mitodl/open-edx-plugins/blob/ab334ee18d7f4eb026e1cd5b64760e4c3b15a0f5/run_devstack_integration_tests.sh) file.
   - Open the terminal and execute the command `./run_devstack_integration_tests.sh`.
